### PR TITLE
Feature/search controller

### PR DESCRIPTION
**Pull request by DavidWrightOS**
# Explore View Controller

Started setting up the explore view controller UI, and present it upon successful login

Trello URL: https://trello.com/c/uAAXAMus
PR Branch: https://github.com/Lambda-School-Labs/LabsPT13-Resfeber-A-IOS/tree/feature/search-controller

### CODE REVIEW

**Comment by joshrutkowski** [comment]
ExploreViewController looking great!

**Comment by joshrutkowski** [comment]
[ExploreViewController initial set up. Create cards to see a list of destinations (using mock data)](https://trello.com/c/7PPsjR0N/51-exploreviewcontroller-initial-set-up-create-cards-to-see-a-list-of-destinations-using-mock-data)

File: Resfeber/View Controllers/ExploreViewController.swift [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        let buttonDiameter: CGFloat = 32
+        button.setDimensions(height: buttonDiameter, width: buttonDiameter)
+        button.layer.cornerRadius = buttonDiameter / 2
+        button.layer.masksToBounds = true
+        button.contentMode = .scaleAspectFill
+        button.backgroundColor = .systemGray3
+        button.addTarget(self, action: #selector(profileImageTapped), for: .touchUpInside)
+        
+        let image = UIImage(systemName: "person.fill")!.withTintColor(.systemGray6, renderingMode: .alwaysOriginal)
+        button.setImage(image, for: .normal)
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/ExploreViewController.swift:36]: `button.setImage()` accepts an optional UIImage, so you probably don't need to force unwrap on line 35.

File: Resfeber/View Controllers/ExploreViewController.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    // MARK: - Helpers
+    
+    fileprivate func configureViews() {
+        view.backgroundColor = UIColor.Resfeber.background
+        navigationController?.navigationBar.isHidden = true
+        
+        // Configure Title Label
+        view.addSubview(titleLabel)
+        titleLabel.anchor(top: view.safeAreaLayoutGuide.topAnchor, left: view.leftAnchor,
+                          right: view.rightAnchor, paddingTop: 4, paddingLeft: 20, paddingRight: 20)
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/ExploreViewController.swift:70]: Spacing looks weird here

File: Resfeber/View Controllers/ExploreViewController.swift [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        // Configure Profile Button
+        view.addSubview(profileButton)
+        profileButton.anchor(right: view.rightAnchor, paddingRight: 20)
+        profileButton.centerY(inView: titleLabel)
+        
+        // Configure Search Bar
+        searchBar.delegate = self
+        view.addSubview(searchBar)
+        searchBar.anchor(top: titleLabel.bottomAnchor, left: view.leftAnchor, right: view.rightAnchor,
+                         paddingTop: 12, paddingLeft: 12, paddingRight: 12)
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/ExploreViewController.swift:81]: Check spacing

File: Resfeber/View Controllers/ExploreViewController.swift [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        let queryText = searchText ?? ""
+        print("DEBUG: Perform query with text: \(queryText)..")
+    }
+}
+
+// MARK: - Collection View Layout
+
+extension ExploreViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let numberOfColumns: CGFloat = 2
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/ExploreViewController.swift:110]: Do you always want this to be two? Or with larger size classes, do you want to increase it?

File: Resfeber/View Controllers/ExploreViewController.swift (hunk 1 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/ExploreViewController.swift:36]: Agreed, I will optional chain it instead

File: Resfeber/View Controllers/ExploreViewController.swift (hunk 4 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/ExploreViewController.swift:110]: I do always want it to be 2. For right now we just are developing the app for iPhone only. Landscape orientation will also be disabled.

File: Resfeber/View Controllers/ExploreViewController.swift (hunk 2 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/ExploreViewController.swift:70]: @johnnydhicks A few lines, such as this one, are rather long so I chose to manually break it up across two lines. I know there are some style guides that have character limits for each line, although I am not following a strict limit here. What is your general recommendation for formatting long lines of code? Is it acceptable to just leave it all on one line and rely on Xcode's editor window auto wrap it?

Comment (johnnydhicks) [reply Resfeber/View Controllers/ExploreViewController.swift:70]: I think if you are going to do line breaks like this, I would do it at every parameter, like this
```titleLabel.anchor(top: view.safeAreaLayoutGuide.topAnchor, 
                                left: view.leftAnchor,
                                right: view.rightAnchor, 
                                paddingTop: 4, 
                                paddingLeft: 20, 
                                paddingRight: 20)

Comment (DavidWrightOS) [reply Resfeber/View Controllers/ExploreViewController.swift:70]: Ok I like that. I'll make the change and submit another PR.

File: Resfeber/View Controllers/ExploreViewController.swift (hunk 3 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/ExploreViewController.swift:81]: I will fix the spacing per your suggestion in the previous comment.

**Pull request merged**